parsl/executors/workqueue/workqueue_worker.py

```diff
@@ -59,15 +57,19 @@ def check_file(parsl_file_obj, mapping, file_type_string):
         input_function = open(input_function_file, "rb")
         function_info = pickle.load(input_function)
```

Comment:
Worth adding a try catch here. If the python versions do not match, this load can fail.

Reply:
Sounds good!  I think that we have to rework this wrapper so that all exceptions are sent back to parsl as they should.

Reply:
See #1688